### PR TITLE
nautilus: pybind/mgr/rbd_support: delay creation of progress module events

### PR DESCRIPTION
**Pull request by smithfarm**
backport tracker: https://tracker.ceph.com/issues/48342

---

backport of https://github.com/ceph/ceph/pull/38196
parent tracker: https://tracker.ceph.com/issues/48296

this backport was staged using ceph-backport.sh version 15.1.1.389
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by smithfarm** [comment]
I had to do this backport entirely manually, so a sanity-check would be welcome.

**Comment by dillaman** [comment]
from a quick check it looks correct -- let me play-test it locally before approving since we don't have test coverage.

**Comment by smithfarm** [comment]
```
1 tests failed out of 178

Total Test time (real) = 3769.11 sec

The following tests FAILED:
	 56 - unittest_lockdep (Timeout)
```

**Comment by smithfarm** [comment]
jenkins test make check

File: src/pybind/mgr/rbd_support/module.py [hunk 1]
```diff
@@ -580,6 +585,10 @@ def to_dict(self):
              TASK_MESSAGE: self.message,
              TASK_REFS: self.refs
              }
+        if self.retry_message:
+            d[TASK_RETRY_MESSAGE] = self.retry_message
+        if self.retry_attempts:
+            d[TASK_RETRY_ATTEMPTS] = self.retry_attempts
```
Comment (dillaman) [inline src/pybind/mgr/rbd_support/module.py:591]: Nit: looks like `TASK_RETRY_ATTEMPTS` is not defined

**Comment by smithfarm** [comment]
@dillaman Indeed, while copying I missed the very first line I was supposed to copy. Added it.

**Review by trociny — APPROVED** [review]
LGTM and seems to work when testing locally.

**Review by dillaman — APPROVED** [review]
:+1:

**Comment by yuriw** [comment]
[wip-yuri4-testing-2021-01-20-0940-nautilus](https://trello.com/c/RoKulpzl/1117-wip-yuri4-testing-2021-01-20-0940-nautilus)

**Pull request merged**